include/aws/common/aws_profile.h

```diff
@@ -0,0 +1,178 @@
+    bool has_profile_prefix;
+};
+
+/**
+ * The profile specification has rule exceptions based on what file
+ * the profile collection comes from.
+ */
+enum aws_profile_source_type { AWS_PST_NONE, AWS_PST_CONFIG, AWS_PST_CREDENTIALS };
+
+struct aws_profile_collection {
+    struct aws_allocator *allocator;
+    enum aws_profile_source_type profile_source;
+    struct aws_hash_table profiles;
+};
+
+AWS_EXTERN_C_BEGIN
+
+/*************************
+ * Profile collection APIs
+ *************************/
+
+/**
+ * Clean up everything associated with a profile collection
+ */
+AWS_COMMON_API
+void aws_profile_collection_destroy(struct aws_profile_collection *profile_collection);
+
+/**
+ * Create a new profile collection by parsing a file with the specified path
+ */
+AWS_COMMON_API
+struct aws_profile_collection *aws_profile_collection_new_from_file(
+    struct aws_allocator *allocator,
+    const struct aws_string *file_path,
+    enum aws_profile_source_type source);
+
+/**
+ * Create a new profile collection by merging a config-file-based profile
+ * collection and a credentials-file-based profile collection
+ */
+AWS_COMMON_API
+struct aws_profile_collection *aws_profile_collection_new_from_merge(
+    struct aws_allocator *allocator,
+    const struct aws_profile_collection *config_profiles,
+    const struct aws_profile_collection *credentials_profiles);
+
+/**
+ * Create a new profile collection by parsing text in a buffer.  Primarily
+ * for testing.
+ */
+AWS_COMMON_API
+struct aws_profile_collection *aws_profile_collection_new_from_buffer(
+    struct aws_allocator *allocator,
+    const struct aws_byte_buf *buffer,
+    enum aws_profile_source_type source);
+
+/**
+ * Retrieves a profile with the specified name, if it exists, from the profile collection
+ */
+AWS_COMMON_API
+struct aws_profile *aws_profile_collection_get_profile(
+    const struct aws_profile_collection *profile_collection,
+    const struct aws_string *profile_name);
+
+/**
+ * Returns how many profiles a collection holds
+ */
+AWS_COMMON_API
+size_t aws_profile_collection_get_profile_count(const struct aws_profile_collection *profile_collection);
+
+/**************
+ * profile APIs
+ **************/
+
+/**
+ * Retrieves a property with the specified name, if it exists, from a profile
+ */
+AWS_COMMON_API
+struct aws_profile_property *aws_profile_get_property(
+    const struct aws_profile *profile,
+    const struct aws_string *property_name);
+
+/**
+ * Returns how many properties a profile holds
+ */
+AWS_COMMON_API
+size_t aws_profile_get_property_count(const struct aws_profile *profile);
+
+/***********************
+ * profile property APIs
+ ***********************/
+
+/**
+ * Returns the value of a sub property with the given name, if it exists, in the property
+ */
+AWS_COMMON_API
+const struct aws_string *aws_profile_property_get_sub_property(
+    const struct aws_profile_property *property,
+    const struct aws_string *sub_property_name);
```

Comment:
document whether the returned string is new, or a reference

trivial, but if it's new, no need to make it `const`